### PR TITLE
python3.12: add `EXTERNALLY-MANAGED` for PEP 668

### PR DESCRIPTION
**Pull request by naveen521kk**
mintty_uiVaiD9ym6

### CODE REVIEW

File: mingw-w64-python3.12/EXTERNALLY-MANAGED [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+Error=To install Python packages system-wide, try 'pacman -S
+ $MINGW_PACKAGE_PREFIX-python-xyz', where xyz is the package you
+ are trying to install.
+
+ If you wish to install a non-MSYS2-packaged Python package,
+ create a virtual environment using 'python -m venv path/to/venv'.
+ Then use path/to/venv/bin/python and path/to/venv/bin/pip.
+
+ If you wish to install a non-MSYS2 packaged Python application,
+ it may be easiest to use 'pipx install xyz', which will manage a
```
Comment (ognevny) [inline mingw-w64-python3.12/EXTERNALLY-MANAGED:11]: do we actually have pipx?

Comment (naveen521kk) [reply mingw-w64-python3.12/EXTERNALLY-MANAGED:11]: Oh, we don't have? I assumed we had it 😅.

I guess we should first package `pipx`.

Comment (naveen521kk) [reply mingw-w64-python3.12/EXTERNALLY-MANAGED:11]: I've created https://github.com/msys2/MINGW-packages/pull/22371

File: mingw-w64-python3.12/EXTERNALLY-MANAGED [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+ virtual environment for you. Make sure you have python-pipx
```
Comment (ognevny) [inline mingw-w64-python3.12/EXTERNALLY-MANAGED:12]:
```suggestion
 virtual environment for you. Make sure you have $MINGW_PACKAGE_PREFIX-python-pipx
```

**Comment by lazka** [comment]
As I said in chat, I'm slightly worried that this will break many CI setups, since venvs in MSYS2 aren't always that nice to work with due to missing binary wheels. Maybe I'm wrong though.

One thing we could do is patch pip (https://github.com/pypa/pip/blob/420435903ff2fc694d6950a47b896427ecaed78f/src/pip/_internal/utils/misc.py#L40) to warn instead of raise for starters (and maybe sleep a minute to be annoying)

**Comment by naveen521kk** [comment]
I guess we can merge this, as we've patched up pip. https://github.com/msys2/MINGW-packages/pull/22382

**Comment by lazka** [comment]
A PR to make the warning an error outside of CI: #27642

**Pull request merged**